app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -63,7 +63,8 @@ def fetch_identity_name_from_token
           return @identity_name_from_token if @identity_name_from_token
 
           @logger.debug(LogMessages::Authentication::AuthnJwt::CheckingIdentityFieldExists.new(token_id_field_secret))
-          @identity_name_from_token = decoded_token[token_id_field_secret]
+          token_path = token_id_field_secret.split('/')
+          @identity_name_from_token = decoded_token.dig(*token_path)
```

Comment:
Maybe that part fetching a value of internal token claim should be seperated to a class that this class and validate_restrictions_one_to_one both use?

Reply:
@nessiLahav @tzheleznyak I've took split and dig procedure to a separated class